alibi_detect/utils/warnings.py

```diff
@@ -0,0 +1,24 @@
+import functools
+import warnings
+
+
+def deprecated_alias(**aliases):
+    def deco(f):
+        @functools.wraps(f)
+        def wrapper(*args, **kwargs):
+            rename_kwargs(f.__name__, kwargs, aliases)
+            return f(*args, **kwargs)
+        return wrapper
+    return deco
+
+
+def rename_kwargs(func_name, kwargs, aliases):
+    for alias, new in aliases.items():
+        if alias in kwargs:
+            if new in kwargs:
+                raise TypeError('{} received both {} and {}'.format(
+                    func_name, alias, new))
+            warnings.warn('{} is deprecated; use {}'.format(alias, new),
```

Comment:
I think I would also use backticks around `{}` fields to emphasize that these are names of variables, may need to check best practice though.

Reply:
Also, it seems that the error could be either `ValueError` or `TypeError`, not sure what's best... Semantically I think `ValueError` fits better.

Reply:
Good idea. Looking at the behaviour of `UnboundLocalError` it seems like this error uses `'{}'`, but then [here](https://www.python.org/dev/peps/pep-0572/#appendix-c-no-changes-to-scope-semantics) backticks are returned. I reckon we go with backticks as we use that to represent variables in docstrings in various places...